aws/resource_aws_instance_test.go

```diff
@@ -4170,3 +4189,40 @@ resource "aws_subnet" "test" {
 }
 `, rName)
 }
+
+// must be >= m3 and have an encrypted root volume to eanble hibernation
+const testAccInstanceConfigHibernation = `
+resource "aws_vpc" "foo" {
+	cidr_block = "10.1.0.0/16"
+	tags = {
+		Name = "terraform-testacc-instance-hibernation"
+	}
+}
+
+resource "aws_subnet" "foo" {
+	cidr_block = "10.1.1.0/24"
+	vpc_id = "${aws_vpc.foo.id}"
+	tags = {
+		Name = "tf-acc-instance-hibernation"
+	}
+}
+
+resource "aws_instance" "foo" {
+	ami = "${aws_ami_copy.encrypted_ami.id}"
+	instance_type = "m3.medium"
+	subnet_id = "${aws_subnet.foo.id}"
+	hibernation = "true"
+}
+
+resource "aws_ami_copy" "encrypted_ami" {
+  name              = "terraform-testacc-encrypted-ami"
+  description       = "An encrypted AMI for Terraform acceptance testing"
+  source_ami_id     = "ami-01e24be29428c15b2"
+  encrypted         = "true"
+  source_ami_region = "us-west-2"
+
+  tags {
+    Name = "terraform-testacc-instance-hibernation"
+  }
+}
```

Comment:
Since the introduction of this pull request, EC2 now supports launching encrypted EBS volumes from unencrypted AMIs. Will refactor this on merge. 👍